Sources/MockoloFramework/Utils/TypeParser.swift

```diff
@@ -303,34 +303,103 @@ public struct Type {
+        if typeName.hasPrefix(String.observableVarPrefix) || typeName.hasPrefix(String.rxObservableVarPrefix),
+            let range = typeName.range(of: String.observableVarPrefix), let lastIdx = typeName.lastIndex(of: ">") {
+            let typeParamStr = typeName[range.upperBound..<lastIdx]
+
+            var subjectKind = ""
+            var underlyingSubjectType = ""
+            if let overrideTypes = overrides {
+                if let val = overrideTypes[overrideKey], val.hasSuffix(String.subjectSuffix) {
+                    subjectKind = val
+                } else if let val = overrideTypes["all"], val.hasSuffix(String.subjectSuffix) {
+                    subjectKind = val
+                }
+            }
+
+            if subjectKind.isEmpty {
+                subjectKind = String.publishSubject
+            }
+            underlyingSubjectType = "\(subjectKind)<\(typeParamStr)>"
+
+            var underlyingSubjectTypeDefaultVal: String? = nil
+            if subjectKind == String.publishSubject {
+                underlyingSubjectTypeDefaultVal = "\(underlyingSubjectType)()"
+            } else if subjectKind == String.replaySubject {
+                underlyingSubjectTypeDefaultVal = "\(underlyingSubjectType).create(bufferSize: 1)"
+            } else if subjectKind == String.behaviorSubject {
+                if let val = Type(String(typeParamStr)).defaultSingularVal(isInitParam: isInitParam, overrides: overrides, overrideKey: overrideKey) {
+                    underlyingSubjectTypeDefaultVal = "\(underlyingSubjectType)(value: \(val))"
+                }
+            }
+            return (underlyingSubjectType, underlyingSubjectTypeDefaultVal)
+        }
+        return (nil, nil)
+    }
     
-    func defaultSingularVal(isInitParam: Bool = false) -> String? {
+    private func parseDefaultVal(isInitParam: Bool, overrides: [String: String]?, overrideKey: String = "") -> String? {
+        let arg = self
+        if let val = defaultSingularVal(isInitParam: isInitParam, overrides: overrides, overrideKey: overrideKey) {
+            return val
+        }
+
+        if !arg.isSingular {
+            return nil
+        }
+
+        if arg.hasClosure {
+            return nil
+        }
+
+        let ret = arg.tupleComponents
+        var vals = [String]()
+        for sub in ret {
+            if sub == "," || sub == ":" || sub == "(" || sub == ")" || sub == "=" || sub == " " || sub == "" {
+                vals.append(sub)
+            } else {
+                if let val = Type(sub).defaultSingularVal(isInitParam: isInitParam, overrides: overrides, overrideKey: overrideKey) {
+                    vals.append(val)
+                } else {
+                    return nil
+                }
+            }
+        }
+
+        if !vals.isEmpty {
+            var ret = vals.joined()
+            ret = ret.replacingOccurrences(of: ",", with: ", ")
+            ret = ret.replacingOccurrences(of: ",  ", with: ", ")
```

Comment:
This is pretty subtle, it would be easy to be confused by this. How about something like `ret.split(",").map { $0.trimmingWhiteSpace }.joined(by: ", ") `?

Reply:
good catch, i meant to do that before but haven't